src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,794 @@
+        /// Holiday details will be looked up for that country as well.
+        /// </summary>
+        /// <param name="catalog">Transform catalog</param>
+        /// <param name="inputColumnName">Input column name</param>
+        /// <param name="columnPrefix">Prefix to add to the generated columns</param>
+        /// <param name="country">Country name to get holiday details for</param>
+        /// <returns><see cref="DateTimeEstimator"/></returns>
+        public static DateTimeEstimator DateTimeTransformer(this TransformsCatalog catalog, string inputColumnName, string columnPrefix, DateTimeEstimator.HolidayList country = DateTimeEstimator.HolidayList.None)
+            => new DateTimeEstimator(CatalogUtils.GetEnvironment(catalog), inputColumnName, columnPrefix, country);
+
+        #region ColumnsProduced static extentions
+
+        internal static Type GetRawColumnType(this DateTimeEstimator.ColumnsProduced column)
+        {
+            switch (column)
+            {
+                case DateTimeEstimator.ColumnsProduced.Year:
+                case DateTimeEstimator.ColumnsProduced.YearIso:
+                    return typeof(int);
+                case DateTimeEstimator.ColumnsProduced.DayOfYear:
+                case DateTimeEstimator.ColumnsProduced.WeekOfMonth:
+                    return typeof(ushort);
+                case DateTimeEstimator.ColumnsProduced.MonthLabel:
+                case DateTimeEstimator.ColumnsProduced.AmPmLabel:
+                case DateTimeEstimator.ColumnsProduced.DayOfWeekLabel:
+                case DateTimeEstimator.ColumnsProduced.HolidayName:
+                    return typeof(ReadOnlyMemory<char>);
+                default:
+                    return typeof(byte);
+            }
+        }
+
+        #endregion
+    }
+
+    /// <summary>
+    /// The DateTimeTransformerEstimator splits up a date into all of its sub parts as individual columns. It generates these fields with a user specified prefix:
+    /// int Year, byte Month, byte Day, byte Hour, byte Minute, byte Second, byte AmPm, byte Hour12, byte DayOfWeek, byte DayOfQuarter,
+    /// ushort DayOfYear, ushort WeekOfMonth, byte QuarterOfYear, byte HalfOfYear, byte WeekIso, int YearIso, string MonthLabel, string AmPmLabel,
+    /// string DayOfWeekLabel, string HolidayName, byte IsPaidTimeOff
+    ///
+    /// You can optionally specify a country and it will pull holiday information about the country as well
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    ///
+    /// ###  Estimator Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | No |
+    /// | Input column data type | Int64 |
+    /// | Output column data type | Columns and types listed in the summary |
+    ///
+    /// The <xref:Microsoft.ML.Transforms.CategoryImputerEstimator> is a trivial estimator and does not need training.
```

Comment:
Why is this reference here?

Reply:
Good catch. Renamed it to DateTimeTransformer.